.github/workflows/sdk-update-api-spec.yaml

```
@@ -0,0 +1,38 @@
+name: SDK Update API Spec
+
+on:
+  schedule:
+    - cron: '0 12 * * 3'
+  workflow_dispatch: {}
+
+jobs:
+  api-spec-update:
+    runs-on: ubuntu-20.04
+    if: ${{ github.repository == 'jellyfin/jellyfin-sdk-kotlin' }}
+    steps:
+      - name: Setup Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: 11
+      - name: Setup Gradle cache
+        uses: actions/cache@v2.1.4
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-java-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-java-gradle-
+            ${{ runner.os }}-java-
+      - name: Update generated Jellyfin API spec
+        uses: technote-space/create-pr-action@v2
+        with:
+          EXECUTE_COMMANDS: ./gradlew --build-cache --no-daemon --info updateApiSpecStable
+          COMMIT_MESSAGE: 'update generated Jellyfin API spec'
+          COMMIT_NAME: 'jellyfin-bot'
+          COMMIT_EMAIL: 'team@jellyfin.org'
+          PR_BRANCH_PREFIX: 'chore/'
+          PR_BRANCH_NAME: 'api-gen-update-${PR_ID}'
+          PR_TITLE: 'chore(api-gen): update generated Jellyfin API spec'
```

```suggestion
          PR_TITLE: 'Update OpenAPI'
```

if possible adding a `to [version]` would be nice but I'm not sure if that's possible

---

I can theoretically get the lated tag from the server repo and use that as the `to <version>`

---

8bf2e1d63b3dbaa02478b1bc0375a9881bfb8640 now includes it

---

You _could_ also read it from one of the generated files: https://github.com/jellyfin/jellyfin-sdk-kotlin/blob/master/jellyfin-api/src/main/kotlin-generated/org/jellyfin/sdk/api/info/ApiConstants.kt

No need to do a HTTP request etc.

---

I could indeed :thinking: let me script a bit

---

Okay, as far as I can tell I can't get this wo work without the 'external' http call. This is due to the sequence of how things have to happen. (i.e. the `updateApiSpecStable` happening inside of the `technote-space/create-pr-action`)
Unless this is a hard blocker I would like to stick to the curl call :thinking:

---

Then I'd rather not add it to the title. The latest release on github is not always the same as the latest spec that is used in the Gradle task so your current approach could end up using the wrong title now.

---

Wouldn't it be possible to execute the gradle command before executing the PR task?

---

not really, since the PR task does a bit more than just create a PR (I will give it a try nonetheless), as it also handles diff check, staging and all.

---

Okay just tested I cant move the Gradle step, HOWEVER I remembered `jq` was a thing so I can redirect the curl twoard the actual `latest` API spec and read the version from there (and correct me if I am wrong that would be the resolved version for the generated class anyway)